src/vstest.console/CommandLine/Executor.cs

```diff
@@ -114,6 +115,24 @@ internal int Execute(params string[] args)
                 exitCode = 1;
             }
 
+            if (!isDiag)
+            {
+                var diag = Environment.GetEnvironmentVariable("VSTEST_DIAG");
```

Comment:
Also I am wondering if it would be better to have a file (or files) with env var constants instead of hardcoding them everywhere. It would help to ensure we use the same thing, would help discoverability and also would help to see patterns we have already used.